### PR TITLE
structmatcher: Add a GomegaMatcher

### PR DESCRIPTION
**Pull request by dsharp-pivotal**
Could use some feedback on if `NegatedFailureMessage` is doing the right thing. Other than the negated failure message test, the added tests are copied from the existing tests.

### CODE REVIEW

File: structmatcher/structmatcher.go [hunk 1]
```diff
@@ -87,3 +87,48 @@ func ExpectStructsToMatchIncluding(expected interface{}, actual interface{}, inc
...
+type Matcher struct {
+	expected        interface{}
+	includingFields []string
+	excludingFields []string
+	mismatches      []string
+}
+
+var _ types.GomegaMatcher = &Matcher{}
+
+func Match(expected interface{}) *Matcher {
```
Comment (khuddlefish) [inline structmatcher/structmatcher.go]: It's a bit confusing to have both this and the other `Match` function called the same thing. Can we think of another name? MatchStruct?

Or maybe we could make separate matchers for `MatchIncluding` or `MatchExcluding`?

I'm looking at [gstruct](https://github.com/onsi/gomega/tree/master/gstruct) for examples

Comment (jmcatamney) [reply structmatcher/structmatcher.go]: Yeah, the namespace collision there is a bit annoying.  `MatchStruct` is probably the simplest change, but `structmatcher.MatchStruct` has a bit of a stutter.  Splitting it into `MatchIncluding`, `MatchExcluding`, and `MatchExactly` would also work.  No strong opinions on which approach to take.

File: structmatcher/structmatcher.go [hunk 2]
```diff
@@ -87,3 +87,48 @@ func ExpectStructsToMatchIncluding(expected interface{}, actual interface{}, inc
...
+	} else if m.excludingFields != nil {
+		m.mismatches = StructMatcher(m.expected, actual, true, false, m.excludingFields...)
+	} else {
+		m.mismatches = StructMatcher(m.expected, actual, false, false)
+	}
+	return len(m.mismatches) == 0, nil
+}
+
+func (m *Matcher) FailureMessage(actual interface{}) (message string) {
+	//return format.Message(actual, "to match struct", m.expected)
```
Comment (jmcatamney) [inline structmatcher/structmatcher.go:121]: Looks like the comment is outdated and should be deleted.

**Comment by jmcatamney** [comment]
Other than the `Match` name issue, functionally everything else looks good to me.

**Pull request merged**